src/reports/library/hdf5_writer.hpp

```diff
@@ -0,0 +1,36 @@
+#include <vector>
+#include <fstream>
+#include <mpi.h>
+#include "hdf5.h"
```

Comment:
Is this file supposed to be in the commit?

Reply:
No, is a requirement of the library-> find_package(HDF5)
I will change to <hdf5.h> for clarity

Reply:
Can you use `HighFive` for the hdf5_writer?  That's what this library uses when possible.

Currently the project doesn't build when HDF5_IS_PARALLEL isn't set because `include_directories (${HDF5_INCLUDE_DIRS})` isn't happening.